addons/mass_mailing_sms/controllers/main.py

```diff
@@ -29,11 +30,43 @@ def _check_trace(self, mailing_id, trace_code):
     @http.route(['/sms/<int:mailing_id>/<string:trace_code>'], type='http', website=True, auth='public')
     def blacklist_page(self, mailing_id, trace_code, **post):
         check_res = self._check_trace(mailing_id, trace_code)
+        country_code = request.geoip.country_code
+        country = request.env['res.country'].search([('code', '=', country_code)], limit=1)
+        sms_number = post.get('sms_number', '').strip()
         if not check_res.get('trace'):
             return request.redirect('/odoo')
+
+        if sms_number:
+            # Sanitize the phone number
+            sanitized_number = phone_validation.phone_format(
+                sms_number,
+                country.code,
+                country.phone_code,
+                force_format='E164',
+                raise_exception=False,
+            ) or sms_number  # Use original if sanitization fails
+
+            trace = check_res['trace'].filtered(lambda r: r.sms_number == sanitized_number)[:1] if sanitized_number else False
+            # Redirect if trace exists
+            if trace:
+                params = werkzeug.urls.url_encode({'sms_number': sanitized_number})
+                return request.redirect(f'/sms/{mailing_id}/unsubscribe/{trace_code}?{params}')
+
+            # Handle unsubscribe errors
+            unsubscribe_error = _('Invalid number %s', sms_number) if not sanitized_number else _('Number %s not found', sanitized_number)
```

Comment:
Hello @datonefaridze  wouldn’t it make more sense to have a single error message, something like "Number is invalid or not found"? Since we're not actually validating whether the number is invalid or simply not found, and the input field is already marked as required, there shouldn’t be a case where sms_number is empty when clicking the unsubscribe button. The only time the number might be absent is when we’re redirected to the blacklist page via the link. What do you think? :)

Reply:
@aath-odoo I think logic is incorrect here, previously we had `tocheck_number` variable which had two conditions `sanitized or sms_number` but here we only check on 1 (you can use anti conjunction rule not X and not Y). can u please correct it? also about the error I think you are right, at least I didn't find the edge cases, so if you want to keep 1 error you can do it, (or wait for reviewer and do it then, or do it now and bring it back if reviewer says it, for me it's okay)

Reply:
Hello @datonefaridze  regarding the logic correction, I believe we’ve already accounted for the sanitized or sms_number condition within the sanitized variable itself no? so we don't need an additional check. All of the code is wrapped inside the if sms_number: block, and since the field is required, there shouldn't be any cases where sms_number is empty, except when the user is redirected to the blacklist page via the link.

Let me know if you’d like further adjustments, but I believe this should cover the edge cases! What do you think?